src/utils.rs

```diff
@@ -18,27 +18,27 @@ pub fn commit_file<P: AsRef<Path>>(
 ) -> Result<(), Error> {
     
     let repo = try!(get_repo_from_path(repo_path));
-
-    let _ = repo.index()
+    let tree_oid = try!(repo.index()
```

Comment:
Agreed, it's ugly. Didn't know how to make the map work before, but figured it out. Lmk if still could be improved. Some/all of the `try!()` statements have to stay, in order to keep `repo` in scope.